subprojects/logging/src/main/java/org/gradle/internal/logging/sink/GroupingProgressLogEventGenerator.java

```diff
@@ -226,6 +240,7 @@ synchronized void flushOutput() {
                     listener.onOutput(header());
                 }
 
+                didHaveOutput = bufferedLogs.size() > 0;
```

Comment:
We want to know if the last forwarded output had any output (except for the header). I don't think it would give the same result.